lms/services/moodle.py

```diff
@@ -137,14 +137,15 @@ def list_files(self, course_id: int):
     def file_exists(self, file_id) -> bool:
         """Check if the file exists in the course."""
         # Moodle file IDs are URLs, but they need the token to be accessible
-        response = self._http.request("HEAD", f"{file_id}&token={self.token}")
+        response = self._http.request(
+            "GET", f"{file_id}&token={self.token}", headers={"Range": "bytes=0-5"}
```

Comment:
The standard check is to look for "%PDF" in the first 1024 bytes of the file. _Most_ PDFs will have these characters at the very start, but there may be files that do not, yet which work in other viewers because.